source/common/router/router.cc

```diff
@@ -1042,7 +1042,11 @@ void Filter::onUpstreamReset(Http::StreamResetReason reset_reason,
   }
 
   const bool dropped = reset_reason == Http::StreamResetReason::Overflow;
-  chargeUpstreamAbort(Http::Code::ServiceUnavailable, dropped, upstream_request);
+  Http::Code error_code = Runtime::runtimeFeatureEnabled(
```

Comment:
shouldn't we only set the error code to 503 if the reset reason is ProtocolError?

Reply:
Absolutely! I was wondering why so many tests started to fail :-)
Added logic to return 502 code only when reset was caused by protocol error from upstream.